app/controllers/CreditsSummaryController.scala

```diff
@@ -85,27 +87,37 @@ class CreditsSummaryController @Inject()(creditsView: CreditsSummary,
                     origin: Option[String] = None)
                    (implicit user: MtdItUser[AnyContent], hc: HeaderCarrier, ec: ExecutionContext): Future[Result] = {
     if (isEnabled(MFACreditsAndDebits)) {
-      creditHistoryService.getCreditsHistory(calendarYear, user.nino).flatMap {
-        case Right(credits) =>
-          val charges = credits
-            .sortBy(_.date.toEpochDay)
-          Future.successful(Ok(creditsView(
-            calendarYear = calendarYear,
-            backUrl = if (isAgent) getAgentBackURL(user.headers.get(REFERER), calendarYear) else getBackURL(user.headers.get(REFERER), origin, calendarYear),
-            isAgent = isAgent,
-            utr = user.saUtr,
-            btaNavPartial = user.btaNavPartial,
-            enableMfaCreditsAndDebits = isEnabled(MFACreditsAndDebits),
-            charges = charges,
-            origin = origin)))
-        case Left(_) => {
-          if (isAgent) {
-            Logger("application").error(s"[CreditsSummaryController][showAgentCreditsSummary] - Could not retrieve financial details for Calendar year: $calendarYear, NINO: ${user.nino}")
-            Future.successful(agentItvcErrorHandler.showInternalServerError())
-          }
-          else {
-            Logger("application").error(s"[CreditsSummaryController][showCreditsSummary] - Could not retrieve financial details for Calendar year: $calendarYear, NINO: ${user.nino}")
-            Future.successful(itvcErrorHandler.showInternalServerError())
+      creditService.getCreditCharges().flatMap { financialDetailsModels =>
+        creditHistoryService.getCreditsHistory(calendarYear, user.nino).flatMap {
+          case Right(credits) =>
+            val charges = credits
+              .sortBy(_.date.toEpochDay)
+
+            val maybeBalanceDetails: Option[BalanceDetails] = financialDetailsModels match {
```

Comment:
this can be moved within creditService and used on the line 96 as it is;

Reply:
Thank you, I've just pushed the fix